brave/brave-ios
# Fix #6706: Add support of SNS navigation.

## Summary of Changes
Brave will now detect user input url is SNS domain
User will need to choose either disable SNS resolver or enable it.
Leave it no chosen will be considered as getting asked every time.
The settings is available under Web3. The default value is Ask
However, everything is behind a feature flag WalletFeatureFlags.SNSDomainResolverEnabled.
Local build can change this value enable to try it out.


This pull request fixes #6706 #6893

## Submitter Checklist:


## Test Plan:

1. Make sure `BraveWalletSns` BraveCore Switches is on (changes can only be applied with a fresh launch)
2. Make sure the feature flag is on
3. Input `onybose.sol` in the omnibox then enter to go to
4. An interstitial page will show up explains what Brave can help with the SNS domain
5. Check copies are correct
6. Click `Disable`
7. observe web page will load and gives an error
8. Go to Settings/Web3/Web3 Domains
9. observe the `Resolve Solana Name Service (SNS) Domain Names` value is `Disabled`
10. chose `Ask` for `Resolve Solana Name Service (SNS) Domain Names`
11. back to the browser and hit reload 
12. observe the interstitial page is back
13. this time click `Proceed`
14. observe the url `onybose.sol` is resolved to `anirudha.co` and loads successfully 
15. Go to Settings/Web3/Web3 Domains
16. observe now he `Resolve Solana Name Service (SNS) Domain Names` value is `Enabled`


## Screenshots:

![simulator_screenshot_52634090-E4A2-4B4B-A419-444B5E7CD248](https://user-images.githubusercontent.com/1187676/217415250-20c86884-a070-4b4b-b180-d97eea8c5594.png)


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
